include/iocore/cache/HttpTransactCache.h

```diff
@@ -30,8 +30,9 @@
  ****************************************************************************/
 #pragma once
 
-#include "tscore/ink_platform.h"
+#include "proxy/hdrs/HTTP.h"
```

Comment:
We made an effort to avoid including `proxy/` from `iocore/cache/` before. 
https://github.com/apache/trafficserver/commit/5932a30e3f10956abf9d88cf4e57ee28bc20f1f4

Reply:
I agree that we should avoid including `proxy/` from `iocore/`, but as it stands today, HTTPHdr is dereferenced in HttpTransactCache.cc, for example, [here](https://github.com/moonchen/trafficserver/blob/b7b16275d210577b08d90db870570f718b9eecc7/src/iocore/cache/HttpTransactCache.cc#L200).

Reply:
I guess `HttpTransactCache.cc` can include the `proxy/` headers by itself and this `HttpTransactCache.h` header file doesn't need to include?

Reply:
As talked at bug scrub, it looks like this just uncovered existing circular dependency. Let's fix it after shipping this.